### DISCUSSION TITLE
Multiple HAdevices on one physical device

### DISCUSSION THREAD

**Discussion by jribenfors** [dawidchyrzynski/arduino-home-assistant#232] [Q&A]
Is it possible to have one microprocessor running several devices?

I'm planning multiple sensors sending data to a base device over lora. The base device would create HADevices for each remote sensor  - based on the ID in the packet - and send them on to Home Assistant. Each remote sensor might equate to one device, or multiple remote sensors might be combined as multiple types on a single device.

Is this possible with this library? It looks like there can only be one instance of the HAMqtt class, and that expects only one device when it's created.

**Comment by dawidchyrzynski** [comment]
Hi @jribenfors, the current implementation does not support having multiple HADevices on a single physical device. I may consider adding this feature in the future, but I am unable to provide any ETA for its implementation.
